### DISCUSSION TITLE
Cypress taking too long when running in CI pipeline

### DISCUSSION THREAD

**Discussion by onimougwo** [cypress-io/cypress#16344] [CI setup]
Hi everyone,

My cypress tests are running in about **7 minutes** on my local machine pointing to a static site in Azure.

When they run the Azure CI pipeline on a build controller (on-prem) they take **14 minutes** pointing to that exact same static site in Azure.

I also tried to run the tests from an Azure VM to that same static site and they take **14 minutes**. 

Why such a big gap ? I checked the CPU and the RAM on the build controller and the Azure VM and it's not bad so what other elements could contribute to the slowness? 

See benchmark in attachment 

Thanks 

![Cypress_execution](https://user-images.githubusercontent.com/40011061/117164586-c234c200-ad92-11eb-8c46-255886a81989.png)

**Comment by lmiller1990** [comment]
How many tests do you have? Will need some more information here. How long do they take to run in other environments (eg locally)?

What version of Cypress? Could you try using older versions (eg, 7.x, 6.9, 6.8...) and see if the perf/speed is a regression.
